### DISCUSSION TITLE
关于V神的QAP文章

### DISCUSSION THREAD

**Discussion by idanieltam** [SutuLabs/zkcourse#4] [Week1]
[链接](https://vitalik.ca/general/2016/12/10/qap.html)

```
t = [-88.0, 592.666, -1063.777, 805.833, -294.777, 51.5, -3.444]
Z = [24, -50, 35, -10, 1]
h = t / Z = [-3.666, 17.055, -3.444] with no remainder
```

请问老师或看过的同学，这里的division是什么运算？ h为什么是这个vector并没有余数

**Comment by wizicer** [comment]
https://qiita.com/kzee/items/4666993bd2f9790a0642

这篇文章里面有详细描述整个计算过程（相同的数字，更详细的解释），试试看看？

**Comment by readygo67** [comment]
这里的division 是多项式除法
t(x)= −3.440556∗x^6+51.471∗x^5−294.720056∗x^4+805.7885∗x^3−1063.749889∗x^2+592.652∗x−88.
z(x)= x^4 - 10*x^3 + 35*x^2-50*x +24 
h(x) = -3.444*x^2+17.055*x -3.666

因为x=1/2/3/4是t(x)=0的解，所以如果对t(x) 做因式分解，一定有(x-1)(x-2)(x-3)(x-4)这样的因式， 所以t(x) 一定能被z(x) 整除。
